mobile_packet_verifier/src/banning/db.rs

```diff
@@ -0,0 +1,109 @@
+use chrono::{DateTime, Duration, Utc};
+use file_store::mobile_ban::{BanType, VerifiedBanReport};
+use futures::TryStreamExt;
+use helium_crypto::PublicKeyBinary;
+use sqlx::{PgConnection, Row};
+
+use super::{BannedRadios, BAN_CLEANUP_DAYS};
+
+// When retreiving banned radios, we exclude Poc bans.
+// They still exist in the db to make updating bans easier.
+pub async fn get_banned_radios(
+    pool: &mut PgConnection,
+    timestamp: DateTime<Utc>,
+) -> anyhow::Result<BannedRadios> {
+    let banned = sqlx::query(
+        r#"
+        SELECT hotspot_pubkey
```

Comment:
This should only get the most recent record batching the time window per hostpost?

Reply:
Updated with [b45ab77](https://github.com/helium/oracles/pull/984/commits/b45ab7769712e00619fd6a1a85b6eb2d8b84ae3d)

We get a `DISTINCT ON (hotspot_pubkey)` with `ORDER BY hotspot_pubkey, received_timestamp DESC`.
This gives us the latest row for a hotspot at any given time. 
Then we filter by `WHERE ban_type IN ('all', 'poc')`.